examples/ChatQnA/deploy/gaudi.md

```diff
@@ -0,0 +1,974 @@
+  -X POST \
+  -d '{"inputs":"What is Deep Learning?", \
+     "parameters":{"max_new_tokens":17, "do_sample": true}}' \
+  -H 'Content-Type: application/json'
+
+```
+
+TGI service generate text for the input prompt. Here is the expected result from TGI:
+
+```
+{"generated_text":"We have all heard the buzzword, but our understanding of it is still growing. It’s a sub-field of Machine Learning, and it’s the cornerstone of today’s Machine Learning breakthroughs.\n\nDeep Learning makes machines act more like humans through their ability to generalize from very large"}
+```
+
+**NOTE**: After launch the TGI, it takes few minutes for TGI server to load LLM model and warm up.
+:::
+::::
+
+
+If you get
+
+```
+curl: (7) Failed to connect to 100.81.104.168 port 8008 after 0 ms: Connection refused
+
+```
+
+and the log shows model warm up, please wait for a while and try it later.
+
+```
+2024-06-05T05:45:27.707509646Z 2024-06-05T05:45:27.707361Z  WARN text_generation_router: router/src/main.rs:357: `--revision` is not set
+2024-06-05T05:45:27.707539740Z 2024-06-05T05:45:27.707379Z  WARN text_generation_router: router/src/main.rs:358: We strongly advise to set it to a known supported commit.
+2024-06-05T05:45:27.852525522Z 2024-06-05T05:45:27.852437Z  INFO text_generation_router: router/src/main.rs:379: Serving revision bdd31cf498d13782cc7497cba5896996ce429f91 of model Intel/neural-chat-7b-v3-3
+2024-06-05T05:45:27.867833811Z 2024-06-05T05:45:27.867759Z  INFO text_generation_router: router/src/main.rs:221: Warming up model
+
+```
+
+### LLM Microservice
+
+```
+curl http://${host_ip}:9000/v1/chat/completions\
+  -X POST \
+  -d '{"query":"What is Deep Learning?","max_new_tokens":17,"top_k":10,"top_p":0.95,\
+     "typical_p":0.95,"temperature":0.01,"repetition_penalty":1.03,"streaming":true}' \
+  -H 'Content-Type: application/json'
+
+```
+
+You will get generated text from LLM:
+
+```
+data: b'\n'
+data: b'\n'
+data: b'Deep'
+data: b' learning'
+data: b' is'
+data: b' a'
+data: b' subset'
+data: b' of'
+data: b' machine'
+data: b' learning'
+data: b' that'
+data: b' uses'
+data: b' algorithms'
+data: b' to'
+data: b' learn'
+data: b' from'
+data: b' data'
+data: [DONE]
+```
+
+### MegaService
+
+```
+curl http://${host_ip}:8888/v1/chatqna -H "Content-Type: application/json" -d '{
+     "model": "Intel/neural-chat-7b-v3-3",
+     "messages": "What is the revenue of Nike in 2023?"
+     }'
+
+```
+
+Here is the output for your reference:
+
+```
+data: b'\n'
+data: b'An'
+data: b'swer'
+data: b':'
+data: b' In'
+data: b' fiscal'
+data: b' '
+data: b'2'
+data: b'0'
+data: b'2'
+data: b'3'
+data: b','
+data: b' N'
+data: b'I'
+data: b'KE'
+data: b','
+data: b' Inc'
+data: b'.'
+data: b' achieved'
+data: b' record'
+data: b' Rev'
+data: b'en'
+data: b'ues'
+data: b' of'
+data: b' $'
+data: b'5'
+data: b'1'
+data: b'.'
+data: b'2'
+data: b' billion'
+data: b'.'
+data: b'</s>'
+data: [DONE]
+```
+##### Gaurdrail Microservice

```

Comment:
This heading is H5 while the previous one was H3.   Should this be inside the previous heading (hence H4, with only four #) or at the same level (H3)?

Also Guardrail is misspelled

Reply:
you are right